lib/src/booleans/boolean_disjoint.dart

```diff
@@ -0,0 +1,125 @@
+
+import '../../helpers.dart';
+import '../../meta.dart';
+import '../line_intersect.dart';
+import '../polygon_to_line.dart';
+import 'boolean_point_in_polygon.dart';
+
+/// Returns [true] if the intersection of the two geometries is an empty set.
+/// example:
+/// ```dart
+/// var point = Point(coordinates: Position.of([2, 2]));
+/// var line = LineString(
+///   coordinates: [
+///     Position.of([1, 1]),
+///     Position.of([1, 2]),
+///     Position.of([1, 3]),
+///     Position.of([1, 4])
+///   ],
+/// );
+/// booleanDisjoint(line, point);
+/// //=true
+/// ```
+bool booleanDisjoint(GeoJSONObject feature1, GeoJSONObject feature2) {
+  var bool = true;
+  flattenEach(
+    feature1,
+    (flatten1, featureIndex, multiFeatureIndex) {
+      flattenEach(
+        feature2,
+        (flatten2, featureIndex, multiFeatureIndex) {
+          if (!bool) {
+            return bool;
+          }
+          bool = _disjoint(flatten1.geometry!, flatten2.geometry!);
+        },
+      );
+    },
+  );
+  return bool;
+}
+
+/// Disjoint operation for simple Geometries ([Point]/[LineString]/[Polygon])
+bool _disjoint(GeometryType geom1, GeometryType geom2) {
+  if (geom1 is Point) {
+    if (geom2 is Point) {
+      return geom1.coordinates != geom2.coordinates;
+    } else if (geom2 is LineString) {
+      return !isPointOnLine(geom2, geom1);
+    } else if (geom2 is Polygon) {
+      return !booleanPointInPolygon((geom1).coordinates, geom2);
+    }
+  } else if (geom1 is LineString) {
+    if (geom2 is Point) {
+      return !isPointOnLine(geom1, geom2);
+    } else if (geom2 is LineString) {
+      return !isLineOnLine(geom1, geom2);
+    } else if (geom2 is Polygon) {
+      return !isLineInPoly(geom2, geom1);
+    }
+  } else if (geom1 is Polygon) {
+    if (geom2 is Point) {
+      return !booleanPointInPolygon((geom2).coordinates, geom1);
+    } else if (geom2 is LineString) {
+      return !isLineInPoly(geom1, geom2);
+    } else if (geom2 is Polygon) {
+      return !_isPolyInPoly(geom2, geom1);
+    }
+  }
+  return false;
+}
+
+// http://stackoverflow.com/a/11908158/1979085
+bool isPointOnLine(LineString lineString, Point pt) {
+  for (var i = 0; i < lineString.coordinates.length - 1; i++) {
+    if (isPointOnLineSegment(lineString.coordinates[i],
+        lineString.coordinates[i + 1], pt.coordinates, true)) {
+      return true;
+    }
+  }
+  return false;
+}
+
+bool isLineOnLine(LineString lineString1, LineString lineString2) {
+  var doLinesIntersect = lineIntersect(lineString1, lineString2);
+  if (doLinesIntersect.features.isNotEmpty) {
+    return true;
+  }
+  return false;
+}
+
+bool isLineInPoly(Polygon polygon, LineString lineString) {
+  for (var coord in lineString.coordinates) {
+    if (booleanPointInPolygon(coord, polygon)) {
+      return true;
+    }
+  }
+  var doLinesIntersect = lineIntersect(lineString, polygonToLine(polygon));
+  if (doLinesIntersect.features.isNotEmpty) {
+    return true;
```

Comment:
btw is there a reason, why these are not private? Are they already shared between functions? Do they need to be exposed?